props/saber_fett263_buttons.h

```diff
@@ -257,7 +331,96 @@ SaberFett263Buttons() : PropBase() {}
       case EVENTID(BUTTON_POWER, EVENT_CLICK_SHORT, MODE_OFF | BUTTON_AUX):
         previous_preset();
         return true;
+
+       // Gesture Controls
+		
+       // Battle Mode
+		    
+#ifdef FETT263_BATTLE_MODE
+       case EVENTID(BUTTON_NONE, EVENT_SWING, MODE_ON | BUTTON_AUX):
+         if (!battle_mode_) {
+           battle_mode_ = true;
+           hybrid_font.SB_Force();
+         } else {
+           battle_mode_ = false;
+           hybrid_font.SB_Force();
+         }
+         return true;
+
+       // Auto Lockup Mode
+       case EVENTID(BUTTON_NONE, EVENT_CLASH, MODE_ON):
+         if (!battle_mode_) return false;
+         clash_impact_millis_ = millis();
+         swing_blast_ = false;
+         if (!swinging_) {
+           SaberBase::SetLockup(SaberBase::LOCKUP_NORMAL);
+           swing_blast_ = false;
+           auto_lockup_on_ = true;
+           SaberBase::DoBeginLockup();
+           return true;
+	 }
+
+       case EVENTID(BUTTON_NONE, EVENT_STAB, MODE_ON):
+         if (!battle_mode_) return false;
+         clash_impact_millis_ = millis();
+         swing_blast_ = false;
+         if (!swinging_) {
+           if (fusor.angle1() < - M_PI / 4) {
+             SaberBase::SetLockup(SaberBase::LOCKUP_DRAG);
+             } else {
+               SaberBase::SetLockup(SaberBase::LOCKUP_MELT);
+       	     }
+         swing_blast_ = false;
+         auto_melt_on_ = true;
+         SaberBase::DoBeginLockup();
+         return true;
+         }
 	
+       case EVENTID(BUTTON_NONE, EVENT_SWING, MODE_OFF):  
+         if(millis() > 3000) { 
+           On();
+           battle_mode_ = true;
+         }
+         return true;
+
+       case EVENTID(BUTTON_NONE, EVENT_TWIST, MODE_ON):
```

Comment:
So let's see if I get this right....

if you enable FETT263_BATTLE_MODE, then you use swing to turn the saber on, and twist to turn it off?
While I think using swing to turn it on is kind of cool, it seems like a feature that is completely separate from battle mode to me.  Assuming battle mode works well, it would make the saber work a lot more like it does in the movies, but swing on/twist off is not how it works in the movies....

Anyways, this is completely up to you, but I think it would be cool if swing-on was not required for battle mode.
For my purposes, I would probably want a define that turns on battle mode all the time instead of having it be a setting.

Reply:
Battle Mode has two activation methods, if the blade is on you can hold Aux and Swing to toggle on/off, if the blade is off and you use any gesture ignition it is also turned on (Swing On, Twist On or Stab On all activate) - the idea is if you use a gesture to ignite the controls automatically switch to the gesture "Battle Mode" if you use the power button it will be regular unless you activated battle mode previously but you can enter or exit Battle Mode anytime with Aux + Swing.
The Twist Off is a "safety" feature in Battle Mode to prevent accidental retraction, you can also just exit Battle Mode any press Power to turn off.  
I'm keeping it a "mode" at this time because it essentially removes Stab and Clash effects but if testing proves out then we can add a define to have it always on easily enough, because it's a significant change I wanted to take one step at a time.  Personally, I will keep it as a mode for my use but if users just want it default it can certainly be accomodated.  The controls can still be modified as more users test and give feedback as well, I'm certainly open to more input, these were the best controls based on what was available and what felt most natural while testing.
I'll make other fixes later today.

Reply:
This is exactly why we have prop files though. You can implement this any way you want to, and I can implement it any way I want to in my prop file.... :)

Reply:
For alpha testing I propose it just as a "mode", then from there determine best way to allow users to have it be default.  I was originally trying to make it so you could have it be either a mode or always on with defines but I couldn't get it to work so I will probably need some assistance in having defines "share" functionality.  I want to get the functionality tested and proven then having it be a mode or always on once it's working optimally we can look at along with making a few other controls user defined.  I was hitting some obstacles so probably need your expertise to take it to that level.